augur/parse.py

```diff
@@ -143,6 +142,8 @@ def register_parser(parent_subparsers):
     parser.add_argument('--sequences', '-s', required=True, help="sequences in fasta or VCF format")
     parser.add_argument('--output-sequences', required=True, help="output sequences file")
     parser.add_argument('--output-metadata', required=True, help="output metadata file")
+    parser.add_argument('--output-id-field', required=False, default=DEFAULT_ID_COLUMNS, nargs="+",
+                        help="The record field to use as the sequence identifier in the FASTA output.")
```

Comment:
Based on the number of accepted arguments, this implementation is different from the [`augur curate` counterpart](https://github.com/nextstrain/augur/blob/d2ef34675fb22085ae288636e43e4b457c09d97b/augur/curate/__init__.py#L89-L90) which can be potentially confusing. I suggest maintaining consistency for options of the same name across subcommands. Some ideas:

1. Update `augur parse` to accept a single name, e.g. `default="str", nargs="?"`
2. Update `augur curate` to accept multiple names, e.g. `default=DEFAULT_ID_COLUMNS, nargs="+"`
3. Use a different name, such as `--output-id-fields` to match the plurality of `--metadata-id-columns`

Reply:
I am inclined towards option 1 primarily because the user is required to provide a `--fields col1 col2 strain accession` argument in the same invocation. It seems reasonable to expect the user to choose a specific field name for the ID.

In cases where users omit selecting a field ID name (or perhaps make a typo in the field ID name 😆), `augur parse` will default to searching the standard [DEFAULT_ID_COLUMNS](https://github.com/nextstrain/augur/blob/df15043d6d5aeb0682ead5f738252ff859fed55b/augur/io/metadata.py#L18), which, in this instance, these columns are set to `('strain', 'name')`.

However, I am interested in hearing others' opinions, particularly regarding the pros and cons of option 2. Or if option 1 would be an acceptable choice for now?

Reply:
That sounds reasonable to me! I would go further and suggest that if `--output-id-field` is not present in `--fields`, that should raise an error instead of falling back to `DEFAULT_ID_COLUMNS`.

We can also keep the existing behavior of using the first column if none are found.

So something like:

```py
if args.output_id_field:
    # Check that it is in args.fields
    strain_key = args.output_id_field
else:
    for field in DEFAULT_ID_COLUMNS:
        if field in args.fields:
            strain_key = field
            break
    strain_key = fields[0]
```

Reply:
@victorlin Why do you set `strain_key` to the first entry in `fields` after breaking from the for loop in your example above? Is that a holdover from copy/paste?

Reply:
Oops, I forgot a condition. Edited.

Reply:
Thanks @victorlin and @huddlej! Implemented the modifications and performed a rebase:  https://github.com/nextstrain/augur/pull/1403/commits/8367e59f04a1bffc9e7167ec060c32cffb8c4882. 

Additionally, wrote a test to ensure an error is generated for non-existent fields (e.g. `--output-id-field "notexist"`). This is ready for review or further suggestions.